openwisp_notifications/templates/admin/openwisp_users/user/change_form_object_tools.html

```diff
@@ -3,6 +3,8 @@
 {% load i18n admin_urls %}
 
 {% block object-tools-items %}
-  <li><a class="button" href="{% url 'notifications:user_notification_preference' object_id %}">Notification Preferences</a></li>
+  {% if request.user.is_staff %}
```

Comment:
This should not check the logged-in user, but the user object for which the change page is being rendered. The user belonging to the `object_id`.